### PR TITLE
Fix CriticalAddonsOnly node-taint caused job pend

### PR DESCRIPTION
**Pull request by c3y1huang**
Relate to https://github.com/rancher/rke2/issues/487.

### CODE REVIEW

File: pkg/helm/controller.go [hunk 1]
```diff
@@ -283,6 +283,12 @@ func job(chart *helmv1.HelmChart) (*batch.Job, *core.ConfigMap, *core.ConfigMap)
 				Value:    "true",
 				Effect:   "NoSchedule",
 			},
+			{
+				Key:      "CriticalAddonsOnly",
+				Operator: core.TolerationOpEqual,
+				Effect:   "NoExecute",
+				Value:    "true",
```
Comment (brandond) [inline pkg/helm/controller.go:290]: nit: match the field order from the other tolerations
```suggestion
				Value:    "true",
				Effect:   "NoExecute",
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: pkg/helm/controller.go (hunk 1 above)
Comment (c3y1huang) [reply pkg/helm/controller.go]: Thanks, addressed.

**Comment by c3y1huang** [comment]
Not sure why the CI is failing.

**Comment by brandond** [comment]
CI should probably be converted over from rke to k3s, it's got some odd flakes at the moment.

**Review by galal-hussein — APPROVED** [review]
LGTM, I believe that tests are failing because of the stable repo update

**Review by brandond — APPROVED** [review]

**Pull request merged**